DeusEx/Classes/DXRKillBobPage.uc

```diff
@@ -1,31 +1,79 @@
 class DXRKillBobPage expands DXRBase;
 
-var config class<ScriptedPawn> BobPageClass;
-var config float minDistance;// minimum distance away from any Teleporter with a tag, or PlayerStart
+var config string BobPageClass;
+var config float minDistance;// minimum distance away from any Teleporter, or PlayerStart
+var config string endgamemap;
 var ScriptedPawn BobPage;
 
 function CheckConfig()
 {
     if( config_version == 0 ) {
-        BobPageClass = class'BobPage';
-        minDistance = 1000;
+        BobPageClass = "BobPage";
+        minDistance = 5000;
+        endgamemap = "99_Endgame4.dx";
     }
     Super.CheckConfig();
 }
 
 function FirstEntry()
 {
-    local PathNode p;
-    local string maps[128];
-    local int i, numMaps;
+    local MapExit exit;
+    local NavigationPoint p;
+    local int i ,slot, num;
+    local string map;
     Super.FirstEntry();
 
-    class'DXRTestAllMaps'.static.GetAllMaps(maps);
+    dxr.SetSeed( dxr.seed + dxr.Crc(Class.Name) );
 
-    foreach AllActors(class'PathNode', p) {
-        if(p.name == 'PathNode497') {
-            BobPage = Spawn(BobPageClass,,, p.Location );
+    map = class'DXRTestAllMaps'.static.PickRandomMap(dxr);
+    l("Bob Page map " $ map);
+    if( Caps(map) != Caps(dxr.localURL) )
+        return;
+
+    foreach AllActors(class'NavigationPoint', p) {
+        if( IsGoodBossLocation(p.Location) == false ) continue;
+        num++;
+    }
+
+    slot = rng(num);
+    num = 0;
+    foreach AllActors(class'NavigationPoint', p) {
+        if( IsGoodBossLocation(p.Location) == false ) continue;
+        if( num == slot ) {
+            BobPage = Spawn( class<ScriptedPawn>(GetClassFromString(BobPageClass, class'ScriptedPawn')),,, p.Location );
+            BobPage.bImportant = true;
+            BobPage.BindName = "EndGameBoss";
             break;
         }
+        num++;
+    }
+
+    SetTimer(1.0, true);
+}
+
+function Timer()
+{
+    Super.Timer();
+    if( dxr.flags.f.GetBool('EndGameBoss_Dead') ) {
+        dxr.player.ConsoleCommand("open " $ endgamemap);
```

Comment:
don't need to use the console command anymore, DXRHordeMode does it better with `Level.Game.SendPlayer(dxr.player, "11_PARIS_CATHEDRAL");`